modules/ml/src/ann_mlp.cpp

```diff
@@ -1390,23 +1648,106 @@ class ANN_MLPImpl : public ANN_MLP
 };
 
 
+
+class ANN_MLP_ANNEALImpl : public ANN_MLPImpl
```

Comment:
This is not need as separate class. Merge this contents into ANN_MLPImpl.
- revert "=0" change in .hpp file (but keep "const" fix).
- remove these new calls with dynamic_cast: `double ANN_MLP_ANNEAL::getAnnealInitialT() const`